newrelic/core/application.py

```diff
@@ -1238,6 +1245,28 @@ def harvest(self, shutdown=False, flexible=False):
                                 data_sampler.name,
                             )
 
+                    # Send environment plugin list
+
+                    stopwatch_start = time.time()
+                    while (
+                        configuration
+                        and configuration.package_reporting.enabled
+                        and self._env_send
+                        and ((time.time() - stopwatch_start) < 2.0)
```

Comment:
I'm not sure if I mentioned this already or not but some agents have a configurable rate limit for this instead of a hardcoded value. That being said could we place 2.0 at the top of the file as a constant so it's not a magic number here? Something like:
```
MAX_PACKAGE_CAPTURE_TIME_PER_SLOW_HARVEST = 2.0
```